crawl-ref/source/dat/des/variable/large_themed.des

```diff
@@ -2316,3 +2316,100 @@ xxxx-xx..xx....xx-xxx-x
 xxxxxxxxx-x.....xxxxxxx
 xxxxxxxxxxx.@@@..xxxxxx
 ENDMAP
+
+NAME:    dreamdust_isle_of_ice
+TAGS:    no_monster_gen
+ORIENT:  south
+DEPTH:   D:11-, Lair
+WEIGHT:  3
+SUBST:   - = .....wwwW
+SUBST:   ~ = wwwwwW
+COLOUR:  . = white
+FTILE:   .x = floor_ice
+RTILE:   x = wall_ice_block
+COLOUR:  x = blue
+: set_feature_name("shallow_water", "Some ice crusted shallow water")
+: set_feature_name("deep_water", "Some ice crusted deep water")
+: set_feature_name("rock_wall", "ice covered rock wall")
+KMONS:   . = nothing w:200 / ice beast / polar bear / \
+             wolf w:5 / white imp w:5 / rime drake w:3 / \
+             ice devil w:3 / ice statue w:2
+MAP
+              WWWWWWWWWWWWW
+           WWWWWWWWWWWWWWWWWWW
+       WWWWWW~~~~~~~~~~~~~~~WWWWW
+    WWWWWW~~~~----.....----~~~~WWWWWW
+  WWWW~~~~~---.............---~~~~~WWWW
+ WWW~~~----...................----~~~WWW
+WW~~~----.......................----~~~WW
+~~~xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx~~~
+xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+ENDMAP
+
+NAME:   dreamdust_labyrinth
+TAGS:   no_monster_gen no_trap_gen
+ORIENT: northeast
+DEPTH:  D:11-, !D:$, Lair, !Lair:$
+WEIGHT: 1
+: if crawl.coinflip() then
+# Start in the middle
+NSUBST: ? = 1:( / *:G
+NSUBST: - = 1:< / *:-
+: else
+# Start on the outside
+NSUBST: - = 1:( / *:-
+NSUBST: ? = 1:< / *:G
+: end
+# Place a few dead-ends
+NSUBST: - = 1:x / *:-
+NSUBST: . = 1:x / *:.
+NSUBST: ` = 1:x / *:`
+NSUBST: " = 1:X / *:"
+# Make a path
+NSUBST: f = 1:' / *:f
+NSUBST: e = 1:' / *:e
+NSUBST: d = 1:' / *:d
+NSUBST: c = 1:' / *:c
+NSUBST: b = 1:' / *:b
+# Everything else
+KFEAT:  abcx = stone_wall
+KFEAT:  deX = metal_wall
+KFEAT:  f = crystal_wall
+TILE:   zabcdx = wall_lab_stone
+TILE:   deX = wall_lab_metal
+FTILE:  -.`"'T<G = floor_gauntlet
+KFEAT:  < = escape_hatch_up
+KMONS:  < = patrolling minotaur ; any armour . \
```

Comment:
Use of `any armour` and `any weapon` won't work well for monster equipment. For `any armour`, the minotaur may get equipment it can't actually use, and the item will never get created. For `any weapon`, it may get a launcher with no ammunition. You might look into using the (dgn.monster_weapon)[https://github.com/crawl/crawl/blob/master/crawl-ref/source/dat/dlua/dungeon.lua#L915] function with a class like "knight" or "warrior" and then just defining some good potential armours:
```lua
kmons("< = patrolling minotaur ; " .. dgn.monster_weapon("knight", nil, "good_item") .. 
    " . plate armour | chain mail | scale mail")
``` 
You can get fancier and add in a shield (which won't get used if the minotaur gets a 2h weapon), give it some javelins, random wands etc. It might work best if you just pick a few alternate items for each "slot", which can be weighted, and you can have an entry for each slot of "nothing" as well.

Reply:
I tried looking into this briefly, but I think I may be too dumb. Haha
I ended up just making it a regular patrolling minotaur after doing other tweaks to the vault (the mino is now always in the center, so having javelins shouldn't matter as much as when it could previously generate on the outer perimeter of the vault). Hope that's ok.